mods/bucket/init.lua

```diff
@@ -225,3 +225,19 @@ minetest.register_craft({
 	replacements = {{"bucket:bucket_lava", "bucket:bucket_empty"}},
 })
 
+-- Register buckets as dungeon loot
+if dungeon_loot and dungeon_loot.register then
+	local loot_list = {
+		{name = "bucket:bucket_empty", chance = 0.55},
+		-- water in deserts/ice or above ground, lava otherwise
+		{name = "bucket:bucket_water", chance = 0.45,
+			types = {"sandstone", "desert", "ice"}},
+		{name = "bucket:bucket_water", chance = 0.45, y = {0, 32768},
+			types = {"normal"}},
+		{name = "bucket:bucket_lava", chance = 0.45, y = {-32768, -1},
+			types = {"normal"}},
+	}
+	for _,loot in pairs(loot_list) do
+		dungeon_loot.register(loot)
+	end
```

Comment:
you can register lists of loot, there is no need for the for loop

Reply:
Done for farming as well.